git/config.py

```diff
@@ -251,7 +276,17 @@ def __init__(self, file_or_files, read_only=True, merge_includes=True):
         if not hasattr(self, '_proxies'):
             self._proxies = self._dict()
 
-        self._file_or_files = file_or_files
+        if file_or_files is not None:
+            self._file_or_files = file_or_files
+        else:
+            if config_level is None:
+                if read_only:
+                    self._file_or_files = [get_config_path(f) for f in CONFIG_LEVELS[:-1]]
```

Comment:
I don't think it's an issue, we already have code code from the Repo object (below) which seems to work. If there are any issues, they would have already happened. If anything, it'll affect speed since it would attempt to read a value from the same file twice instead of just once.

```
class Repo(object):
...
    def config_reader(self, config_level=None):
        """docs"""
        files = None
        if config_level is None:
            files = [self._get_config_path(f) for f in self.config_level]
        else:
            files = [self._get_config_path(config_level)]
        return GitConfigParser(files, read_only=True)
...
```

If you'd like to deal with it, I think there should be a deduper in the ConfigParser instead of a fix on that line and on the corresponding line in Repo, although either method would work.

Reply:
Thanks a lot! I wasn't aware this was the same issue traditionally. Apparently it didn't impede function, thus I am happy to accept continued reduced performance in some cases on windows.